tools/d2g/d2g.go

```diff
@@ -102,6 +105,32 @@ func Scopes(dwScopes []string) (gwScopes []string) {
 			gwScopes[i] = s
 		}
 	}
+
+	if dwTaskDef == nil {
+		return
+	}
+
+	var parsedDwTaskDef map[string]interface{}
+	err = json.Unmarshal(dwTaskDef, &parsedDwTaskDef)
+	if err != nil {
+		return []string{}, fmt.Errorf("cannot parse Docker Worker task definition: %v", err)
+	}
+
+	dwPayload := new(dockerworker.DockerWorkerPayload)
+	defaults.SetDefaults(dwPayload)
+	dwPayloadJSON, err := json.Marshal(parsedDwTaskDef["payload"])
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal Docker Worker payload: %v", err)
+	}
+	err = json.Unmarshal(dwPayloadJSON, &dwPayload)
+	if err != nil {
+		return nil, fmt.Errorf("cannot unmarshal Docker Worker payload: %v", err)
+	}
+
+	if dwPayload.Capabilities.Devices.KVM {
+		gwScopes = append(gwScopes, fmt.Sprintf("generic-worker:os-group:%s/kvm", parsedDwTaskDef["taskQueueId"]))
+	}
```

Comment:
i think this ends up being the only part you need to keep, once you pass in a native docker worker payload go type

Reply:
`taskQueueId` is not in the payload though. That's why I passed in the entire task def.